src/transformers/modeling_squeezebert.py

```diff
@@ -0,0 +1,1032 @@
+]
+
+
+def transpose_x(x):
+    return x.permute(0, 2, 1)  # [N, W, C] <--> {N, C, W]
+
+
+ACT2FN = {"gelu": gelu, "relu": torch.nn.functional.relu}
+
+
+class SqueezeBertEmbeddings(nn.Module):
+    """Construct the embeddings from word, position and token_type embeddings."""
+
+    def __init__(self, config):
+        super().__init__()
+        self.word_embeddings = nn.Embedding(config.vocab_size, config.embedding_size, padding_idx=config.pad_token_id)
+        self.position_embeddings = nn.Embedding(config.max_position_embeddings, config.embedding_size)
+        self.token_type_embeddings = nn.Embedding(config.type_vocab_size, config.embedding_size)
+
+        # self.LayerNorm is not snake-cased to stick with TensorFlow model variable name and be able to load
+        # any TensorFlow checkpoint file
+        self.LayerNorm = BertLayerNorm(config.hidden_size, eps=config.layer_norm_eps)
+        self.dropout = nn.Dropout(config.hidden_dropout_prob)
+
+        # position_ids (1, len position emb) is contiguous in memory and exported when serialized
+        self.register_buffer("position_ids", torch.arange(config.max_position_embeddings).expand((1, -1)))
+
+    def forward(self, input_ids=None, token_type_ids=None, position_ids=None, inputs_embeds=None):
+        if input_ids is not None:
+            input_shape = input_ids.size()
+        else:
+            input_shape = inputs_embeds.size()[:-1]
+
+        seq_length = input_shape[1]
+
+        if position_ids is None:
+            position_ids = self.position_ids[:, :seq_length]
+
+        if token_type_ids is None:
+            token_type_ids = torch.zeros(input_shape, dtype=torch.long, device=self.position_ids.device)
+
+        if inputs_embeds is None:
+            inputs_embeds = self.word_embeddings(input_ids)
+        position_embeddings = self.position_embeddings(position_ids)
+        token_type_embeddings = self.token_type_embeddings(token_type_ids)
+
+        embeddings = inputs_embeds + position_embeddings + token_type_embeddings
+        embeddings = self.LayerNorm(embeddings)
+        embeddings = self.dropout(embeddings)
+        return embeddings
+
+
+class MatMulWrapper(torch.nn.Module):
+    """
+    Wrapper for torch.matmul(). This makes flop-counting easier to implement.
+    Note that if you directly call torch.matmul() in your code, the flop counter will typically
+    ignore the flops of the matmul.
+    """
+
+    def __init__(self):
+        super().__init__()
+
+    def forward(self, mat1, mat2):
+        """
+
+        :param inputs: two torch tensors
+        :return: matmul of these tensors
+
+        Here are the typical dimensions found in BERT (the B is optional)
+            mat1.shape: [B, <optional extra dims>, M, K]
+            mat2.shape: [B, <optional extra dims>, K, N]
+            output shape: [B, <optional extra dims>, M, N]
+        """
+        return torch.matmul(mat1, mat2)
+
+
+BertLayerNorm = nn.LayerNorm  # This uses NWC data layout
```

Comment:
@LysandreJik - do we still do this `nn.LayerNorm` renaming? I never fully understood why we are renaming here

Reply:
Indeed, we don't do that anymore! I think originally the `torch` implementation of `LayerNorm` was different to the ones we were using, and as PyTorch versions evolved their `LayerNorm` became identical to ours.

We've kept the aliases but have started removing them a few weeks ago because they make no sense anymore.

Reply:
Yes to remove those and use `nn.LayerNorm`.